src/components/forms/DatePicker/DatePicker.tsx

```diff
@@ -24,20 +24,21 @@ import {
 } from './utils'
 import { Calendar } from './Calendar'
 
-interface DatePickerProps {
+export interface DatePickerProps {
   id: string
   name: string
   className?: string
   disabled?: boolean
   required?: boolean
   defaultValue?: string
-  minDate: string
+  minDate?: string
   maxDate?: string
   rangeDate?: string
   onChange?: (val?: string) => void
   onBlur?: (
     event: React.FocusEvent<HTMLInputElement> | React.FocusEvent<HTMLDivElement>
   ) => void
+  inputProps?: JSX.IntrinsicElements['input']
```

Comment:
I meant to add a note about this change. I assume this is a breaking change, so I'm looking at alternatives as well.

Since the `DateRangePicker` uses `DatePickerProps` to pass through props to each `DatePicker`, I wanted `DateRangeProps` to include all the props in `DatePicker`'s signature. 

One problem I noticed while keeping the intersection type is that the custom `onChange` prop of `DatePicker` conflicts with `onChange` from JSX.IntrinsicElements['input']:

```
Type '(val?: string | undefined) => void' is not assignable to type '((val?: string | undefined) => void) & ((event: ChangeEvent<HTMLInputElement>) => void)'
```

Reply:
For example, the following fixes both issues (prevents breaking changes, and avoids the "onChange" overlap), while still allowing DateRangePicker to only need to know about the exported `DatePickerProps` for passing through to each `DatePicker`:

```
interface BaseDatePickerProps {
  id: string
  name: string
  className?: string
  disabled?: boolean
  required?: boolean
  defaultValue?: string
  minDate?: string
  maxDate?: string
  rangeDate?: string
  onChange?: (val?: string) => void
  onBlur?: (
    event: React.FocusEvent<HTMLInputElement> | React.FocusEvent<HTMLDivElement>
  ) => void
}

export type DatePickerProps = BaseDatePickerProps & Omit<JSX.IntrinsicElements['input'], "onChange">

export const DatePicker = (
  props: DatePickerProps
): React.ReactElement => {
  const {
    id,
    name,
    defaultValue,
    disabled,
    required,
    minDate = DEFAULT_MIN_DATE,
    maxDate,
    rangeDate,
    onChange,
    onBlur,
    ...inputProps
  } = props
```

Reply:
Having`BaseDatePickerProps` and then `DatePickerProps` as the intersection with `input` types works. I think we did something like that with `<CharacterCount /> . I'm fine with using `Omit` for now as way to get around the issue with types... but it points to something bigger.

We probably need to be strict with what is an `onChange`. This is is an issue in Combobox as well (which I helped work on haha)  - we have `onChange` handling the value of the input but not the actual ChangeEvent like the native HTML. I don't know what to do about this... I wonder if for v2 we should have clearer naming of our props so that `onChange` is only ever handling `ChangeEvent` and anything else is called `onValueChange` or something like that...

Reply:
Updated in de11085

I like the idea of having separate verbiage for onChange vs similar custom implementations!